src/main/java/tc/oc/pgm/commands/MapCommands.java

```diff
@@ -46,10 +49,19 @@ public static PGMMap peekNextMap() {
         return map;
       }
     }
+
+    Iterator<Match> iterator = PGM.getMatchManager().getMatches().iterator();
+    PGMMap current = iterator.hasNext() ? iterator.next().getMap() : null;
+
     List<PGMMap> maps = new ArrayList<>(PGM.get().getMapLibrary().getMaps());
-    Collections.shuffle(maps);
-    setNextMap(maps.get(0));
-    return nextMap.get();
+    PGMMap next;
+    do {
+      Collections.shuffle(maps);
+      next = maps.get(0);
+    } while (maps.size() > 1 && Objects.equals(current, next));
```

Comment:
We typically haven't used the `do..while` convention in PGM, can we just use `while`?

Reply:
do/while and while are two different types of loops. do..while is the simplest way to loop at least once. A while condition will be complex here.

Reply:
Ok, that's fine